### PR TITLE
sorter: reduce memory malloc to avoid too much CPU overhead

### PR DESCRIPTION
**Pull request by amyangfei**
### What problem does this PR solve? 

ref: #1853

### What is changed and how it works?

~Since the access of `h.heap` is not concurrent, we can use the same `slice`~

The heap will be used in an async pool. We choose a conservative initialize size for sort heap slice.

- the heap is flushed every 100ms or accumulated data size execeeds `1GB`?

### Check List 

Tests 

 - Unit test
 - Integration test

### Release note

- Reduce memory malloc in sort heap to avoid too much CPU overhead.

### CODE REVIEW

File: cdc/puller/sorter/heap_sorter.go [hunk 1]
```diff
@@ -159,7 +160,7 @@ func (h *heapSorter) flush(ctx context.Context, maxResolvedTs uint64) error {
 			return nil
 		}
 		oldHeap = h.heap
-		h.heap = make(sortHeap, 0, 65536)
+		h.heap = make(sortHeap, 0, sortHeapCapacity)
```
Comment (lonng) [inline cdc/puller/sorter/heap_sorter.go:163]: Do we need to allocate a new slice for `heap`? Can we reuse the previous one, eg: `h.heap = h.heap[:0]`

Comment (liuzix) [reply cdc/puller/sorter/heap_sorter.go:163]: No. Flushing the heap to disk is done in a separate goroutine, so we need to keep the original underlying array intact until it's been flushed.

**Pull request merged**